pootle/apps/pootle_store/models.py

```diff
@@ -1972,7 +1971,7 @@ def update(self, overwrite=False, store=None, only_newer=False, user=None,
                                                        update_unitids, user,
                                                        revision=update_revision,
                                                        submission_type=submission_type)
-            self.file_mtime = disk_mtime
+            self.file_mtime = self.get_file_mtime() or self.file_mtime
```

Comment:
> I'm not either sure I understand the purpose of the or self.file_mtime part. Independently of its meaning, I'm concerned it's a detail being taken care of in a ad-hoc fashion — easy to miss out.

@julen im not sure i follow - do you have any specific suggestions?

the bigger question for me is should a store that previously had a file but now doesn't, set its file_mtime back to datetime_min or should it keep the file_mtime that it had.

 Im not sure it matters that much as the comparison in update_stores at least is only whether file_mtime has changed not whether its newer.